libraries/lib-numeric-formats/NumericConverterRegistry.cpp

```diff
@@ -0,0 +1,147 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/**********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file NumericConverterRegistry.cpp
+
+ Dmitry Vedenko
+
+ **********************************************************************/
+#include "NumericConverterRegistry.h"
+
+namespace
+{
+const auto PathStart = L"NumericConverterRegistry";
+
+struct RegistryVisitor : public Registry::Visitor
+{
+   RegistryVisitor(
+      NumericConverterRegistry::Visitor _visitor, NumericConverterType requestedType)
+       : visitor { std::move(_visitor) }
+       , requestedType { std::move(requestedType) }
+   {
+   }
+
+   void BeginGroup(Registry::GroupItem& item, const Path&) override
+   {
+      auto concreteGroup = dynamic_cast<NumericConverterRegistryGroup*>(&item);
+
+      mInMatchingGroup =
+         concreteGroup != nullptr && concreteGroup->type == requestedType;
+   }
+
+   void EndGroup(Registry::GroupItem&, const Path&) override
+   {
+      mInMatchingGroup = false;
```

Comment:
There is an assumption that groups don't nest.

Reply:
There is indeed. BeginGroup assumes it as well.